plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts

```diff
@@ -544,6 +452,175 @@ export const initProductScreenTracks = () => {
 				} );
 			}
 		} );
+};
+
+/**
+ * Adds product variations tracks.
+ */
+const addProductVariationsTracks = () => {
+	document
+		.querySelector(
+			'#variable_product_options_inner .variations-add-attributes-link'
+		)
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_variations_empty_state', {
+				action: 'add_attribute_link',
+			} );
+		} );
+
+	document
+		.querySelector(
+			'#variable_product_options_inner .variations-learn-more-link'
+		)
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_variations_empty_state', {
+				action: 'learn_more_link',
+			} );
+		} );
+
+	const variationsSection = '#variable_product_options';
+
+	// We attach the events in this way because the buttons are added dynamically.
+	attachEventListenerToParentForChildren( variationsSection, [
+		{
+			eventName: 'click',
+			query: '.generate_variations',
+			callback: () => {
+				recordEvent( 'product_variations_buttons', {
+					action: 'generate_variations',
+				} );
+			},
+		},
+		{
+			eventName: 'click',
+			query: '.add_variation_manually',
+			callback: () => {
+				recordEvent( 'product_variations_buttons', {
+					action: 'add_variation_manually',
+				} );
+			},
+		},
+		{
+			eventName: 'change',
+			query: '#field_to_edit',
+			callback: () => {
+				const selectElement = document.querySelector(
+					'#field_to_edit'
+				) as HTMLSelectElement;
+				// Get the index of the selected option
+				const selectedIndex = selectElement.selectedIndex;
+				recordEvent( 'product_variations_buttons', {
+					action: 'bulk_actions',
+					selected: selectElement.options[ selectedIndex ]?.value,
+				} );
```

Comment:
The idea sounds good, but we would be recording twice the bulk actions that require that confirmation dialog. And if we would want to avoid that situation, we should handle every option individually to add the tracks event, depending on what the user selects.
I honestly feel that it's not worth doing that change now, since we only will "win" knowing whether the user confirmed the selected action.
In order to accomplish the changes described in [the associated Epic](https://github.com/woocommerce/woocommerce/issues/37100) in time, what do you think about creating a new issue to tackle that improvement in the next cycle?

Reply:
A new issue to improve this is okay.